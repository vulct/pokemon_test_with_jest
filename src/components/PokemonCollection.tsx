import React, { FC } from "react";
import { Pokemon } from "../interface";
import PokemonList from "./PokemonList";
import "./pokemon.css";

interface Props {
  pokemons: Pokemon[];
}

const PokemonCollection: FC<Props> = (props) => {
  const { pokemons } = props;

  return (
    <div>
      <h2>Pokemon List</h2>
      <section className="collection-container">
        {pokemons.map((pokemon) => {
          
          return <PokemonList 
          key={pokemon.id}
          id={pokemon.id}
          name={pokemon.name}
          image={pokemon.sprites.front_default}
          />;
        })}
      </section>
    </div>
  );
};

export default PokemonCollection;
