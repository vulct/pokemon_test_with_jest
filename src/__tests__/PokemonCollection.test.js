import { shallow } from "enzyme";
import PokemonCollection from "../components/PokemonCollection";

describe("PokemonCollection", () => {
  let wrapper;

  beforeEach(() => {
    const testItems = [
      {
        id: 1,
        name: "bob",
        sprites: {
          front_default: "123",
        },
      },
      {
        id: 2,
        name: "james",
        sprites: {
          front_default: "456",
        },
      },
    ];

    wrapper = shallow(<PokemonCollection pokemons={testItems} />);
  });

  it("Render <PokemonCollection>", () => {
      expect(wrapper).toMatchSnapshot();
  });

  it("length of collection should", () => {
    expect(wrapper.find(".collection-container")).toHaveLength(1);
  });
  
});
