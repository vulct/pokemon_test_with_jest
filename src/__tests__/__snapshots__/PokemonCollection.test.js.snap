// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonCollection Render <PokemonCollection> 1`] = `
<div>
  <h2>
    Pokemon List
  </h2>
  <section
    className="collection-container"
  >
    <PokemonList
      id={1}
      image="123"
      key="1"
      name="bob"
    />
    <PokemonList
      id={2}
      image="456"
      key="2"
      name="james"
    />
  </section>
</div>
`;
